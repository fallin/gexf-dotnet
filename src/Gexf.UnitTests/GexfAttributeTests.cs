using System.IO;
using System.Xml.Linq;
using NUnit.Framework;
using Shouldly;

namespace Gexf.UnitTests
{
    [TestFixture(TestOf = typeof(GexfAttribute))]
    public class GexfAttributeTests
    {
        [Test]
        public void ToXmlShouldRenderMultipleOptionsWithCorrectDelimiter()
        {
            var subject = new GexfAttribute("id", "title", GexfDataType.String)
            {
                Options = new GexfOptions { "value1", "value2" }
            };

            var output = new GexfOutput();
            var document = subject.ToXml(output);

            var writer = new StringWriter();
            document.Save(writer, SaveOptions.OmitDuplicateNamespaces);
            
            writer.ToString().ShouldMatchApproved();
        }
    }
}