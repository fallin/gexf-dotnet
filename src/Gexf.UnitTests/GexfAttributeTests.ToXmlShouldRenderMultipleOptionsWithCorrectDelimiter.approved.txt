<?xml version="1.0" encoding="utf-16"?>
<attribute id="id" title="title" type="string" xmlns="http://www.gexf.net/1.2draft">
  <options>value1|value2</options>
</attribute>